crates/shared/src/price_estimation/buffered.rs

```diff
@@ -92,7 +90,7 @@ impl PriceEstimating for BufferingPriceEstimator {
                 let remaining_queries: Vec<_> =
                     remaining_queries.iter().flatten().cloned().collect();
                 let inner = self.inner.clone();
-                async move { inner.estimator.estimates(&remaining_queries).await }
+                async move { vec_estimates(inner.estimator.as_ref(), &remaining_queries).await }
```

Comment:
Is this what we want here? Doesn't this mean that you need to wait for all queries to resolve and only then will you start returning `Poll::Ready` in the stream? In that case, waiting for the first one or all of them has the same delay right?

Reply:
> See if there are smarter PriceEstimating implementations possible for the more complicated buffered, competition, instrumented estimators.

I see, this is already covered in the PR description.